public/main.js

```diff
@@ -0,0 +1,87 @@
+const { app, BrowserWindow } = require('electron');
+const isDev = require('electron-is-dev');
+const path = require('path');
+const os = require('os');
+const bodyParser = require('body-parser');
+const expressApp = require('express')();
+const server = require('http').Server(expressApp);
+const io = require('socket.io')(server);
+
+if (isDev) {
+  console.log('Running in development');
+} else {
+  console.log('Running in production');
+}
+
+let mainWindow;
+
+function createWindow() {
+  expressApp.use(bodyParser.urlencoded({ extended: true }));
+  expressApp.use(bodyParser.json());
+
+  server.listen(3001);
+
+  mainWindow = new BrowserWindow({
+    width: 1200,
+    height: 900,
+    webPreferences: {
+      nodeIntegration: true,
+      webSecurity: false,
+    },
+    icon: './icon_png.png'
+  });
+
+  mainWindow.loadURL(isDev ? 'http://localhost:8080' : `file://${__dirname}/../dist/index.html`);
+
+  if (isDev) {
+    BrowserWindow.addDevToolsExtension(
+      path.join(os.homedir(), './Library/Application Support/Google/Chrome/Default/Extensions/fmkadmapgofadopljbjfkapdkoienihi/3.6.0_0'),
```

Comment:
Why is the directory for this hard coded? Isn't the location going to be different depending on the computer the user has, or is this directory at the very least what's standard for all Macs/Macbooks?

Reply:
In their defense, it is a relative directory, and [fmkadmapgofadopljbjfkapdkoienihi is a thing aparently](https://www.google.com/search?q=fmkadmapgofadopljbjfkapdkoienihi), so this is not necessarily the cause of the issues we assumed it was.

Reply:
Also path.join should be changed with path.resolve()  => Path.join takes all the arguments and ‘normalizes’ them to create a path. Path.join *may not always* result in an absolute path. The resolved path could be a relative file path, and Path.resolve always results in an absolute path.